relayer/Cargo.toml

```diff
@@ -10,6 +10,7 @@ authors = [
 ibc = { path = "../modules" }
 tendermint = "0.15.0"
 tendermint-rpc = { version = "0.15.0", features=["client"] }
+tendermint-proto = "0.1.0"
```

Comment:
Did `rpc` split into `rpc` and `proto`? Just wondering because I'm missing the context here and it is not obvious where this dependency is needed exactly in the relayer (all "use" statements in `relayer` seem unchanged). Thanks!

Reply:
Good point and you found the can of worms we need to agree on!

Tendermint RPC is the RPC library.
Tendermint Proto is the auto-generated structs (from protobuf) from the Tendermint Go definitions. And additionally it contains a trait called "DomainType".

Any `.encode` or `.decode` functions need to know about the trait, hence this library is invoked.

There were multiple options discussed where to store the trait. One idea is to have a DomainType trait in ibc-proto also. Although right now there is a version copied from tendermint-proto, I didn't use it: we want to mix and match tendermint and IBC types in our code and we'd need to invoke separate traits for the two different types. And the two traits are called the same! So it would be confusing, copy-ing code and overall not needed.

The other idea is to separate the DomainType trait into it's own thing (crate). That would allow other developers to use it for their own purposes. Maybe that's the way in the future but it's not really confirmed yet, that it makes total sense.

So, until then, we could just keep it in tendermint-proto and invoke it from there. Since we most probably will use tendermint-proto too for other structs, it's convenient.

Reply:
Thanks for the explanation Greg, that was very instructive! 

> The other idea is to separate the DomainType trait into it's own thing (crate). That would allow other developers to use it for their own purposes. Maybe that's the way in the future but it's not really confirmed yet, that it makes total sense.

Given my understanding of the codebase ATM, this approach sounds preferable to having two  `DomainType` traits, one in a tendermint- plus a separate one in ibc-crate, because the specification of the two traits would be identical across the crates. But the current solution is pretty good so far, let's see down the road.

> So, until then, we could just keep it in tendermint-proto and invoke it from there. Since we most probably will use tendermint-proto too for other structs, it's convenient.

Ay ay.